src/docs/guides/gin.md

```diff
@@ -0,0 +1,91 @@
+2. [From a GitHub repository](#deploy-from-a-github-repo).
+3. [Using the CLI](#deploy-from-the-cli).
+4. [Using a Dockerfile](#use-a-dockerfile).
+
+## One-Click Deploy from a Template
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/new/template/dTvvSf)
+
+We highly recommend that [you eject from the template after deployment](/guides/deploy#eject-from-template-repository) to create a copy of the repo on your GitHub account.
+
+**Note:** You can also choose from a <a href="https://railway.app/templates?q=gin" target="_blank">variety of Gin app templates</a> created by the community.
+
+## Deploy from a GitHub Repo
+
+To deploy a Gin app on Railway directly from GitHub, follow the steps below:
+
+1. Fork the basic <a href="https://github.com/railwayapp-templates/gin" target="_blank">Gin GitHub repo</a>. 
+    - If you already have a GitHub repo you want to deploy, you can skip this step.
+2. Create a <a href="https://railway.app/new" target="_blank">New Project.</a>
+3. Click **Deploy from GitHub repo**.
+4. Select the `gin` or your own GitHub repo.
+    - Railway requires a valid GitHub account to be linked. If your Railway account isn't associated with one, you will be prompted to link it.
+5. Click **Deploy Now**.
+
+Once the deployment is successful, a Railway [service](/guides/services) will be created for you. By default, this service will not be publicly accessible.
+
+To set up a publicly accessible URL for the service, navigate to the **Networking** section in the [Settings](/overview/the-basics#service-settings) tab of your new service and click on [Generate Domain](/guides/public-networking#railway-provided-domain).
+
+<Image src="https://res.cloudinary.com/railway/image/upload/f_auto,q_auto/v1727691646/docs/languages-and-frameworks/gin-production_nvsmvf.png"
+alt="screenshot of the deployed gin service showing a hello world API response on a browser"
+layout="responsive"
+width={2661} height={1019} quality={100} />
+
+## Deploy from the CLI
+
+1. <a href="/guides/cli#installing-the-cli" target="_blank">Install</a> and <a href="/guides/cli#authenticating-with-the-cli" target="_blank">authenticate with the CLI.</a>
+2. Clone the forked <a href="https://github.com/railwayapp-templates/gin" target="_blank">gin GitHub repo</a> and `cd` into the directory. 
+    - You can skip this step if you already have an app directory or repo on your machine that you want to deploy.
+3. Run `railway init` within the app directory to create a new project. 
+4. Run `railway up` to deploy.
+    - The CLI will now scan, compress and upload our fastapi app files to Railway's backend for deployment.
+
+## Use a Dockerfile
+
+1. Clone the forked `gin` repo and `cd` into the directory.
+    - You can skip this step if you already have an app directory or repo on your machine that you want to deploy.
+3. Create a `Dockerfile` in the `gin` or app's root directory.
+4. Add the content below to the `Dockerfile`:
+    ```bash
```

Comment:
better dockerfile -

```dockerfile
# Use the Go 1.23 alpine official image
# https://hub.docker.com/_/golang
FROM golang:1.23-alpine

# Create and change to the /app directory.
WORKDIR /app

# Copy go mod and sum files
COPY go.mod go.sum ./

# Install project dependencies
RUN go mod download

# Copy local code to the container image.
COPY . ./

# Build the app without debug symbols
RUN go build -ldflags="-w -s" -o app

# Run the app on container startup.
ENTRYPOINT ["./app"]
```

Reply:
1. Ah that makes sense. Thanks. 
2.  Exactly!